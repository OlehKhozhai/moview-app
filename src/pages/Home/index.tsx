import React from 'react';
import { useDispatch, useSelector } from 'react-redux';

import Footer from 'components/Footer';
import HomeBanner from 'components/HomeBanner';
import Divider from 'components/_common/Divider';
import MoviesNavbar from 'components/MoviesNavbar';
import MoviesList from 'components/MoviesList';
import { moviesSelector } from 'redux/selectors';
import { getMoviesAction } from 'redux/actions';
import useSearchParams from 'hooks/useSearchParams';

const Home = () => {
  const { movies, isMoviesLoading } = useSelector(moviesSelector);
  const dispatch = useDispatch();
  const searchParams = useSearchParams();

  React.useEffect(() => {
    if (searchParams) {
      dispatch(getMoviesAction({ params: searchParams }));
    }
  }, [searchParams, dispatch]);

  return (
    <>
      <HomeBanner />
      <Divider />
      <MoviesNavbar searchParams={searchParams} />
      <MoviesList movies={movies} isLoading={isMoviesLoading} />
      <Footer />
    </>
  );
};

export default React.memo(Home);
