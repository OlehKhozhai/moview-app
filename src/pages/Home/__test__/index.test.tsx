import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { create } from 'react-test-renderer';

import Home from '../index';

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: jest.fn(),
  useSelector: jest.fn().mockReturnValue({ movies: [], isMoviesLoading: false }),
}));

jest.mock('components/Footer/', () => ({
  __esModule: true,
  default: () => '---Footer---',
}));
jest.mock('components/HomeBanner/', () => ({
  __esModule: true,
  default: () => '---HomeBanner---',
}));
jest.mock('components/_common/Divider', () => ({
  __esModule: true,
  default: () => '---Divider---',
}));
jest.mock('components/MoviesNavbar/', () => ({
  __esModule: true,
  default: () => '---MoviesNavbar---',
}));
jest.mock('components/MoviesList/', () => ({
  __esModule: true,
  default: () => '---MoviesList---',
}));

describe('Home component', () => {
  test('should render Home component by default on home page', () => {
    const component = create(
      <MemoryRouter initialEntries={['/']}>
        <Home />
      </MemoryRouter>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
