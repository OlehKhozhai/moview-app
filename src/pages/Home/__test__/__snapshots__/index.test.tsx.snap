// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should render Home component by default on home page 1`] = `
Array [
  "---HomeBanner---",
  "---Divider---",
  "---MoviesNavbar---",
  "---MoviesList---",
  "---Footer---",
]
`;
